packages/shared/pkg/storage/gcp_multipart_test.go

```diff
@@ -0,0 +1,890 @@
+
+			// Update max concurrent parts
+			for {
+				max := atomic.LoadInt32(&maxConcurrentParts)
+				if current <= max || atomic.CompareAndSwapInt32(&maxConcurrentParts, max, current) {
+					break
+				}
+			}
+
+			// Simulate some processing time to increase chance of concurrency
+			time.Sleep(50 * time.Millisecond) // Increased delay to ensure overlap
+
+			partNum := atomic.AddInt32(&partCalls, 1)
+			body, _ := io.ReadAll(r.Body)
+			receivedParts.Store(int(partNum), string(body))
+
+			w.Header().Set("ETag", fmt.Sprintf(`"etag%d"`, partNum))
+			w.WriteHeader(http.StatusOK)
+
+		case strings.Contains(r.URL.RawQuery, "uploadId"):
+			atomic.AddInt32(&completeCalls, 1)
+			w.WriteHeader(http.StatusOK)
+		}
+	})
+
+	uploader := createTestMultipartUploader(t, handler)
+
+	// Use high concurrency to stress test
+	err = uploader.UploadFileInParallel(t.Context(), testFile, 50)
+	require.NoError(t, err)
+
+	// Verify all calls were made
+	require.Equal(t, int32(1), atomic.LoadInt32(&initiateCalls))
+	require.Equal(t, int32(1), atomic.LoadInt32(&completeCalls))
+	require.True(t, atomic.LoadInt32(&partCalls) > 0)
+	require.True(t, atomic.LoadInt32(&maxConcurrentParts) > 1, "Should have concurrent uploads")
+
+	// Verify content integrity
+	var reconstructed strings.Builder
+	for i := 1; i <= int(atomic.LoadInt32(&partCalls)); i++ {
+		if part, ok := receivedParts.Load(i); ok {
+			reconstructed.WriteString(part.(string))
+		}
+	}
+	require.Equal(t, testContent, reconstructed.String())
+}
+
+func TestMultipartUploader_RandomFailures_ChaosTest(t *testing.T) {
+	tempDir := t.TempDir()
+	testFile := filepath.Join(tempDir, "chaos.txt")
+	testContent := strings.Repeat("chaos test data ", 3200000) // ~100MB file for multiple parts
+	err := os.WriteFile(testFile, []byte(testContent), 0o644)
+	require.NoError(t, err)
+
+	var attemptCount, successCount int32
+	failureRate := 0.3 // 30% failure rate
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch {
+		case r.URL.RawQuery == "uploads":
+			response := InitiateMultipartUploadResult{
+				Bucket:   "test-bucket",
+				Key:      "test-object",
+				UploadID: "chaos-upload-id",
+			}
+			xmlData, _ := xml.Marshal(response)
+			w.WriteHeader(http.StatusOK)
+			w.Write(xmlData)
+
+		case strings.Contains(r.URL.RawQuery, "partNumber"):
+			atomic.AddInt32(&attemptCount, 1)
+
+			// Randomly fail some requests
+			if rand.Float64() < failureRate {
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+
+			atomic.AddInt32(&successCount, 1)
+			partNum := strings.Split(strings.Split(r.URL.RawQuery, "partNumber=")[1], "&")[0]
+			w.Header().Set("ETag", fmt.Sprintf(`"chaos-etag-%s"`, partNum))
+			w.WriteHeader(http.StatusOK)
+
+		case strings.Contains(r.URL.RawQuery, "uploadId"):
+			w.WriteHeader(http.StatusOK)
+		}
+	})
+
+	// Use aggressive retry config for chaos test
+	config := RetryConfig{
+		MaxAttempts:       5,
```

Comment:
can you increase at least to 10, now the test fails ~ 1 in 400 

```suggestion
		MaxAttempts:       10,
```